### DISCUSSION TITLE
WASM Bevy Resource interface

### DISCUSSION THREAD

**Discussion by Faeren0327** [bevyengine/bevy#7523] [Q&A] [answered]
Hello,

May I ask you how to access and change a Bevy Resource using wasm?

Example:
Change a Bevy counter Resource.
#[derive(Resource, Debug, Clone, Default)]
struct Counter {
    count: u8,
}

#[wasm_bindgen]
pub fn add_value(counter: ResMut<Counter>) {
    counter.count = counter.count + 1;
}

Thank you very much!

Best regards,
Mate

**Comment by TuNiman1** [comment, marked as answer]
Hello! 
Basically, you can read and learn it from the unofficial bevy book, but I will explain it here a lil bit.

Firstly, you will need to install the rust support for Wasm, very easily and accessible with a simple: rustup target install wasm32-unknown-unknown


Then, you will need to install a cargo tool for the Wasm server, which also can be done very easily with a simple: cargo install wasm-server-runner


After that, you will need to add it to your cargo.toml, simply add: [target.wasm32-unknown-unknown]
runner = "wasm-server-runner"

Now, you will need to set the env variable for the runner by simply typing: export CARGO_TARGET_WASM32_UNKNOWN_UNKNOWN_RUNNER=wasm-server-runner

Finally, you can run it using: cargo run --target wasm32-unknown-unknown

And of course build it with the wasm-bindgen tool, to generate all needed files to your wasm website, simply type both of these commands:
cargo build --release --target wasm32-unknown-unknown

wasm-bindgen --out-dir ./out/ --target web ./target/


Hope that helped(;
If so, accept the discussion so I'll be able to know(:

**Reply by Faeren0327** [reply]
Hy!

The question was related to how to access a Bevy Resource and not how to build the code.
I was able to build, but then I was not sure if it is possible to access a Bevy Resource somehow from javascript.

Meantime I solved it by using a global variable which can be changed using wasm_bindgen function and then I can read the same variable in Bevy. 
I think it is not possible to directly access a Bevy Resource.

But thanks for your support and answer!
Sorry for the late response.

**Comment by TuNiman1** [comment]
Your code seems to be wrong, if it's the issue so check it again or ask for my help(:
